File: leetcode-algorithms/028. Implement strStr()/solution.py
def strStr(haystack, needle):
	"""
	:type haystack: str
	:type needle: str
	:rtype: int
	"""

	for i in range(len(haystack)+1):
		for j in range(len(needle)+1):
			print(i,j)
			if j==len(needle):
				return i
			if i+j == len(haystack):
				return -1
			if needle[j] != haystack[i+j]:
				break

if __name__ == '__main__':
	haystack = 'aaaaa'
	needle = 'bba'
	res = strStr(haystack,needle)
	print(res)

File: leetcode-algorithms/028. Implement strStr()/solution.cpp
#include <iostream>
#include <string>
using namespace std;

int strStr(string haystack, string needle) {
    for(int i=0;;i++)
    	for(int j=0;;j++)
    	{
    		if(j==needle.size())
    			return i;
    		if(i+j == haystack.size())
    			return -1;
    		if(needle[j]!=haystack[i+j])
    			break;
    	}
}

int main()
{
	string haystack("hello");
	string needle("ll");
	int res = strStr(haystack, needle);
	cout<<res<<endl;
	return 0;
}